docs/core/diagnostics/distributed-tracing.md

```diff
@@ -0,0 +1,229 @@
+           try
+           {
+               //process request ...
+           }
+           finally
+           {
+               //stop activity
+               httpListener.StopActivity(activity, new {context} );
+           }
+       }
+   }
+```
+
+Here is example for how to listen to the Activity events:
+
+```csharp
+    DiagnosticListener.AllListeners.Subscribe(delegate (DiagnosticListener listener)
+    {
+        if (listener.Name == "MyActivitySource")
+        {
+            listener.Subscribe(delegate (KeyValuePair<string, object> value)
+            {
+                if (value.Key.EndsWith("Start", StringComparison.Ordinal))
+                    LogActivityStart();
+                else if (value.Key.EndsWith("Stop", StringComparison.Ordinal))
+                    LogActivityStop();
+            });
+        }
+    }
+```
+
+.NET 5.0 has extended the capability of the distributed tracing to allow the [OpenTelemetry](https://opentelemetry.io/) tracing scenarios, added Sampling capabilities, simplified the tracing coding pattern, and made the listening to the Activity events easier and flexible.
+
+> [!NOTE]
+> To access all added .NET 5.0 capabilities, ensure referencing the [System.Diagnostics.DiagnosticSource](https://www.nuget.org/packages/System.Diagnostics.DiagnosticSource/) NuGet package version 5.0 or later. This package can be used in libraries and apps targeting any supported version of the .NET Framework, .NET Core, and .NET Standard. If targeting .NET 5.0 or later, no need to manually reference the package as it is included in the shared library installed with the .NET SDK.
+
+## Getting Started With Tracing
+
+The applications and the libraries can easily publish tracing data by simply using the <xref:System.Diagnostics.ActivitySource?displayProperty=nameWithType> and the <xref:System.Diagnostics.Activity?displayProperty=nameWithType> classes.
+
+### ActivitySource
+
+First step to publish tracing data is to create instance of the ActivitySource class. The ActivitySource is the class provides APIs to create and start Activity objects and to register ActivityListener objects to listen to the Activity events.
+
+```csharp
+    internal static ActivitySource source = new ActivitySource("MyCompany.MyComponent.SourceName", "v1");
+```
+
+#### Best Practices
+
+- Create the ActivitySource once and store it in a static variable and use that instance as long as needed.
+
+- The source name passed to the constructor has to be unique to avoid the conflicts with any other sources. It is recommended to use hierarchical name contains the parts, the company name, the component name, and the source name. For example, `Microsoft.System.HttpClient.HttpInOutRequests`.
+
+- The version parameter is optional. It is recommended to provide the version in case plan to release multiple versions of the library or the application and want to distinguish between the sources of different versions.
```

Comment:
It is recommended to provide the version in case **you** plan to release multiple versions of the library or the application and want to distinguish between the sources of different versions.